src/vs/workbench/contrib/chat/browser/chatInputPart.ts

```diff
@@ -578,8 +621,9 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 		const toolbarsContainer = elements.inputToolbars;
 		this.chatEditingSessionWidgetContainer = elements.chatEditingSessionWidgetContainer;
 		this.renderAttachedContext();
-		if (this.options.enableImplicitContext) {
-			this._implicitContext = this._register(new ChatImplicitContext());
+
+		if (this.options.enableImplicitContext && !this._implicitContext) {
```

Comment:
Why do we need `&& !this._implicitContext`? render shouldn't be called more than once